src/tests/test_utils/functional_test_utils/src/check_node_type.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "functional_test_utils/check_node_type.hpp"
+
+#include <gtest/gtest.h>
+
+#include "openvino/runtime/exec_model_info.hpp"
+#include "openvino/util/common_util.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+inline std::string setToString(const std::unordered_set<std::string> s) {
+    return std::string("{") + ov::util::join(s) + "}";
+}
+
+void CheckNumberOfNodesWithTypeImpl(std::shared_ptr<const ov::Model> function,
```

Comment:
Just wondering, why was it required to move those functions to a separate file?

Reply:
Thanks @EgorDuplensky for reviewing this PR.
1:  Path: `src/tests/functional/plugin/shared/include/subgraph_tests/` and `src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/` share this infrastructure. Avoid to duplicate codes, just move here, as @iefode 's suggestion.
2: I have a special reason to split `CheckNumberOfNodesWithTypeImpl`, just copy original implementation.